### ISSUE TITLE
PS3Eye turns off a few minutes after the service launches

### ISSUE THREAD

**Issue by baronvonclue** [cboulay/PSMoveService#56]
Hi all,

Not a programmer, just a VR enthusiast short on money. I've managed to get a 32 bit build of the service running, but was unable to get the PS3Eye to connect. The service threw me the following error: `libusb: error [winusbx_claim_interface] could not access interface 0: [87] The parameter is incorrect.` I searched around, but all I could find was the same error for the psmove-unity5 project with no obvious solution.

After some thinking, I realized I was compiling a 32 bit build and you had specified using the CLEye driver for 32 bit. After installing this the service detects the camera and I am able to see the feed in the config test stream, but after 1-2 minutes the service disconnects from the camera and the red LED turns off. If I'm running the test stream, this crashes the service. Below is the output in the service window.

```
[2016-06-09 20:34:21.753680]: main - Starting PSMoveService
Config file name: "C:\Users\Usr\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
[2016-06-09 20:34:21.770700]: bluetooth_get_host_address - Found a bluetooth radio
[2016-06-09 20:34:21.779727]: bluetooth_get_host_address - Retrieved radio info
Config file name: "C:\Users\Usr\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
[2016-06-09 20:34:21.853344]: PS3EyeTracker::open - Opening PS3EyeTracker(USB\VID_1415&PID_2000\0\0, camera_index=0)
CLEyeGetCameraCount() found 0 devices.
CL Eye Driver detected.
CL Eye Driver being used with native DShow. Setting m_index to 0
Attempting cv::VideoCapture::open(index)
Config file name: "C:\Users\Usr\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_opencv_0.json"
Parameter changed via registry. Resetting capture device.
Config file name: "C:\Users\Usr\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_opencv_0.json"
Parameter changed via registry. Resetting capture device.
Config file name: "C:\Users\Usr\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_opencv_0.json"
[2016-06-09 20:34:22.233846]: PS3EyeTracker::getVideoFrameDimensions - Unknown video format for cameraUSB\VID_1415&PID_2000\0\0)
[2016-06-09 20:34:22.247869]: SharedMemory::initialize() - Allocating shared memory: tracker_view_0
[2016-06-09 20:34:22.262910]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSEYE) opened
[2016-06-09 20:34:38.571365]: ClientConnection::start - Starting client connection id 0
[2016-06-09 20:34:38.582392]: ClientConnection::send_connection_info - Sending connection id to client 0
[2016-06-09 20:35:02.000163]: DeviceTypeManager::update_connected_devices - Closing device 0 (PSEYE) since it's no longer in the device list.
```
Debugging with Visual Studio gives me `Unhandled exception at 0x012E90C8 in PSMoveService.exe: 0xC0000005: Access violation reading location 0x00000000.`

Trying to build a 64 bit version throws me lots of errors in Visual Studio, all of the format: `error LNK1112: module machine type 'x64' conflicts with target machine type 'X86'`. I suspect this is user error but I'm not familiar enough with Visual Studio to figure it out, apologies.

**Comment by HipsterSloth** [comment]
That's super interesting the from the logs that it's trying to refresh the camera list. The service isn't suppose to refresh the camera list after the service starts. I'll have to look into that.

Have you tried running the pre-built release here: https://github.com/cboulay/PSMoveService/releases? I'd be curious if that has the same problem.

You definitely don't want to try building 64-bit if you want to use the CLEye driver since the CLEYE dll is 32-bit only. It's possible to build the service in 64-bit, but you have to tweak the cmake script to specify a 64-bit project.

If the CLEye driver still doesn't work with the pre-built binary the other option is to use the libusb based driver. You'll have to uninstall the CLEYE driver and then run through the libusb setup process here: https://github.com/cboulay/PSMoveService/wiki/PSEye-Software-Setup-%28Windows%29

Thanks for testing this stuff out! Having people bang their heads against this has shaken out a bunch of bugs.

**Comment by zelmon64** [comment]
I've been sticking with libusb-win32. When I had one of my PSEyes using CL-Eye Driver, PSMoveService  seemed to have a problem with the other webcams connected. Mind you, that was a while ago so things may have changed. I believe libusb-wn32 performs better anyway.

**Comment by cboulay** [comment]
@baronvonclue This might sound trivial, but many other users have had PSEye-related problems resolved by changing the USB-port to which their camera is connected. Also, for libusb only, switching the driver backing the libusb install (e.g. libusbk vs libusb-win32) sometimes helps.

**Comment by baronvonclue** [comment]
I tried two different PSEyes, both on two different USB ports, and had the same problem, even on the precompiled release.

I uninstalled all the drivers and then started from scratch installing libusb-win32. I've no particular desire to use CLEye,  I just tried it because it looked like libusb wasn't working. Had the same "incorrect parameter" error. Tried winusb, service started, test stream started, lots of pink or purple artefacting. Tried libusbk, same story. Changed back to libusb-win32, now works like a charm with both PSEyes attached concurrently. This is all on the precompiled release, haven't retried the one I built.

I have no idea what the problem was, but it seems to have resolved, at least for now.

@HipsterSloth If I left the service running it would continue to throw that parameter error endlessly, continuously reprinting the message. I don't know if that's helpful or not!

I'll continue testing as able. Is there anywhere else for discussion of testing? Don't want to be raising issues if it's just stuff I'm implementing wrong.

**Comment by zelmon64** [comment]
@baronvonclue Sorry I didn't think the libusbK trick would be relevant for PSMoveService. When using PSMoveAPI with FreePIE I found out that libusbK had to be installed first then it could be replaced by libusb-win32. I had assumed this was due to FreePIE since the PSMoveAPI examples were fine without this trick.

This trick only needs to be done once however which is convenient but makes it hard to tell when it is needed. It only needs to be redone if Windows is restored to an earlier point before it was done. After doing this trick the libusb-win32 and CL-Eye drivers can be uninstalled and reientalled arbitrarily without any issues.

**Comment by HipsterSloth** [comment]
@baronvonclue Yeah error is definitely interesting. I'll have to investigate more as to what might be causing that. This is definitely the right place to discuss testing though. As soon as I have something that I think will fix this issue checked in I'll post an update here.

**Comment by cboulay** [comment]
Simple issues are more than welcome. Closing issues makes us feel good! As long as you don't disappear when the issue's been resolved without telling us that it's been resolved. Then that feels bad, gnawing at our souls.

**Comment by toto5100** [comment]
I have the same problem : at first, psye was not working. Then, i installed libusbK driver, launched the service, it was not working (pure green screen), so i replaced libusbK with libusb-win32. Now it works, but the PSEye turns off after a few minutes, and if i'm using it, the service crashes. When it disconnects without using it, it reconnects after a few minutes, and then disconnects again... It's really strange.

**Comment by baronvonclue** [comment]
I think I'm having the same problem as @toto5100 when I compile the latest commit, but the precompiled binaries don't seem to suffer from this problem. To me it seems like it's a different problem from before, though. I don't have any console messages or error logs yet, will try and get some when I have more time.

**Comment by netbeard** [comment]
I'm having the same issue here, I tried the precompiled release on my system (win10 x64) with libusb installed per the instructions on the wiki, and it throws an error for missing CLEyeMulticam.dll.
I copied the dll to the folder and was able to make the error disappear, but with both single and double Eye cameras, I get it briefly working before it spits out the same errors that Baronvonclue gets.

I'm also unable to compile the latest commit in any configuration, I assume it's user error on my part, however. error LNK1112: module machine type 'x64' conflicts with target machine type 'X86'	C:\psmove\buildstuff\PSMoveService-master\PSMoveService\build\src\psmoveservice\libboost_log-vc120-mt-s-1_59.lib(unhandled_exception_count.obj)	PSMoveService

**Comment by cboulay** [comment]
@netbeard The folder structure (PSMoveService-master) makes me think you downloaded the source as a zip file instead of cloned the repo. As [mentioned in the wiki](https://github.com/cboulay/PSMoveService/wiki/Building-from-source#downloading-the-source), you can't do that. You need to clone the repo recursively (or after cloning do a `git submodule init && git submodule update`) because we use submodules.

I don't know if a submodule error would lead to a boost error, but let's start there. Also, seeing as that error is unrelated to this issue, please post a new issue. I won't be addressing that particular error in this issue anymore.

**Comment by HipsterSloth** [comment]
@baronvonclue @netbeard I just fixed issue #63 related to camera disconnection. Would you guys mind either pulling down latest from master OR downloading Alpha3 from the release and see if this fixes your issue? I suspect it's the same problem.

**Comment by HipsterSloth** [comment]
I haven't heard any reports of this particular problem in the last few days so I'm going to close this issue. Please feel free to re-open it if it's still occurring.
